lom/src/plugins/plugins_common/plugins_helper.go

```diff
@@ -164,70 +184,163 @@ func (periodicDetectionPluginUtil *PeriodicDetectionPluginUtil) Init(pluginName
     }
     periodicDetectionPluginUtil.requestFrequencyInSecs = requestFrequencyInSecs
     periodicDetectionPluginUtil.heartBeatIntervalInSecs = actionConfig.HeartbeatInt
-    periodicDetectionPluginUtil.requestAborted = atomic.Bool{}
-    periodicDetectionPluginUtil.shutDownChannel = make(chan interface{})
+    /* Size of responseChannel should be 2, so that the go routine handling request can be terminated on shutdown if the Request method has already terminated. */
+    periodicDetectionPluginUtil.responseChannel = make(chan *lomipc.ActionResponseData, 2)
     periodicDetectionPluginUtil.requestFunc = requestFunction
     periodicDetectionPluginUtil.shutdownFunc = shutDownFunction
     periodicDetectionPluginUtil.PluginName = pluginName
+    periodicDetectionPluginUtil.detectionRunInfo = DetectionRunInfo{}
+    periodicDetectionPluginUtil.ctx, periodicDetectionPluginUtil.cancelCtxFunc = context.WithCancel(context.Background())
     lomcommon.LogInfo(fmt.Sprintf("Initialized periodicDetectionPluginUtil successfuly for (%s)", pluginName))
     return nil
 }
 
-// TODO: Enhance logic in this method to immediately start ticker, ensure hearbeat and request are not blocked by each other.
-/* This method is promoted to the plugin Periodically invokes detection logic and send heartbeat as well. Honors shutdown when shutdown is invoked on plugin */
+/*
+This method immediately starts heartbeat and request execution.
+This method is promoted to the plugin. Honors shutdown when shutdown is invoked on plugin.
+*/
 func (periodicDetectionPluginUtil *PeriodicDetectionPluginUtil) Request(
     hbchan chan PluginHeartBeat,
     request *lomipc.ActionRequestData) *lomipc.ActionResponseData {
     if request.Timeout > 0 {
         return GetResponse(request, "", "", ResultCodeInvalidArgument, "Invalid Timeout value for detection plugin")
     }
-    detectionTicker := time.NewTicker(time.Duration(periodicDetectionPluginUtil.requestFrequencyInSecs) * time.Second)
+
+    // Publish a heartbeat immediately.
+    pluginHeartBeat := PluginHeartBeat{PluginName: periodicDetectionPluginUtil.PluginName, EpochTime: time.Now().Unix()}
+    hbchan <- pluginHeartBeat
+
+    lomcommon.GetGoroutineTracker().Start(plugin_prefix+periodicDetectionPluginUtil.PluginName, periodicDetectionPluginUtil.handleRequest, request)
     heartBeatTicker := time.NewTicker(time.Duration(periodicDetectionPluginUtil.heartBeatIntervalInSecs) * time.Second)
-    lomcommon.LogInfo(fmt.Sprintf("Timers initialized for plugin (%s)", periodicDetectionPluginUtil.PluginName))
-    isExecutionHealthy := false
+    timeoutLogTicker := time.NewTicker(time.Duration(lomcommon.GetConfigMgr().GetGlobalCfgInt(req_timeout_log_periodicity_secs_key)) * time.Second)
+    defer heartBeatTicker.Stop()
```

Comment:
use lomcommon helper for log periodic.

Reply:
Not using log periodic as discussed.

Reply:
Have added a func test which validates that detection logic is executed immediately right after first execution is finished, in case the first detection logic runs longs.